flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/expressions/converter/DirectConvertRule.java

```diff
@@ -39,220 +41,281 @@
  */
 @Internal
 public class DirectConvertRule implements CallExpressionConvertRule {
+    private static Map<Boolean, DirectConvertRule> cachedInstances = new HashMap<>();
 
-    private static final Map<FunctionDefinition, SqlOperator> DEFINITION_OPERATOR_MAP =
+    public static synchronized DirectConvertRule instance(boolean isBatchMode) {
+        DirectConvertRule instance = cachedInstances.get(isBatchMode);
+        if (instance == null) {
+            instance = new DirectConvertRule();
+            instance.init();
+            instance.initDynamicFunctions(isBatchMode);
+            cachedInstances.put(isBatchMode, instance);
+        }
+        return instance;
+    }
+
+    private final Map<FunctionDefinition, SqlOperator> definitionSqlOperatorHashMap =
             new HashMap<>();
 
-    static {
+    void initDynamicFunctions(boolean isBatchMode) {
+        FlinkSqlOperatorTable.dynamicFunctions(isBatchMode)
+                .forEach(
+                        func -> {
+                            if (func.getName()
+                                    .equalsIgnoreCase(SqlStdOperatorTable.CURRENT_DATE.getName())) {
+                                definitionSqlOperatorHashMap.put(
+                                        BuiltInFunctionDefinitions.CURRENT_DATE, func);
+                            } else if (func.getName()
+                                    .equalsIgnoreCase(SqlStdOperatorTable.CURRENT_TIME.getName())) {
+                                definitionSqlOperatorHashMap.put(
+                                        BuiltInFunctionDefinitions.CURRENT_TIME, func);
+                            } else if (func.getName()
+                                    .equalsIgnoreCase(SqlStdOperatorTable.LOCALTIME.getName())) {
+                                definitionSqlOperatorHashMap.put(
+                                        BuiltInFunctionDefinitions.LOCAL_TIME, func);
```

Comment:
It's better all dynamic functions should be defined here, including CURRENT_TIMESTAMP, LOCALTIMESTAMP, NOW

Reply:
This should be done and also for the FlinkSqlOperatorTable, easier for maintainance